### DISCUSSION TITLE
How to implement a composite device with two gamecontrollers

### DISCUSSION THREAD

**Discussion by mrdudz** [hathach/tinyusb#1471] [Q&A]
Hi!

I need to implement a composite device that (besides a keyboard) presents two game controllers. I am using a raspberry pico board for this.

I used the hid_composite example for this - but no matter what i do, i cant make it present a second game controller (all the other devices work fine).

Now the question is - is this at all possible with tinyusb? Or is it perhaps limited to one device per type in a composite device for some reason? If not, what else could i do wrong? As far as i understood the example, all that would be needed is adding another game controller to ```desc_hid_report```, with a new ID of course, and then it should already present that device to the host system.

Any hints? Thanks

**Comment by mrdudz** [comment]
OK to follow up - to my surprise what i did actually works - but only in Windows! In linux i only see one gamecontroller. That seems very odd.

Ideas?

**Reply by hathach** [reply]
2 game controller may need 2 instances of HIDs, check out [hid_multiple_interface](https://github.com/hathach/tinyusb/tree/master/examples/device/hid_multiple_interface) and change the hid descriptor to gamepad. Keyboard can have its own interface (better) or bundled with one of gamepad, it is up to your design.

**Comment by mrdudz** [comment]
That worked great, thank you very much for the hint!
